lib/chef/knife/core/generic_presenter.rb

```diff
@@ -182,7 +182,7 @@ def extract_nested_value(data, nested_value_spec)
             #   `keys` - want the key named `keys`, not a list of
             #   available keys.
             elsif data.respond_to?(:[])
-              data = data[attr]
+              data = data[attr] ? data[attr] : data.send(attr.to_sym)
```

Comment:
Otherwise we probably need to guard this with a `respond_to?` which would then completely duplicate the next `elsif` statement.